File: src/main/java/jp/artan/japanesefoodmod/common/potion/JapanesePotion.java
package jp.artan.japanesefoodmod.common.potion;

import jp.artan.japanesefoodmod.JapaneseFoodMod;
import net.minecraft.potion.Potion;

import static jp.artan.japanesefoodmod.init.Init.POTIONS;

public class JapanesePotion extends Potion {
    private final String name;

    public JapanesePotion(String name, boolean isBadEffectIn, int liquidColorIn) {
        super(isBadEffectIn, liquidColorIn);
        this.name = name;
        this.setPotionName(this.name);
        this.setRegistryName(JapaneseFoodMod.MODID, this.name);

        POTIONS.add(this);
    }

}

File: src/main/java/jp/artan/japanesefoodmod/common/potion/JapanesePotion.kt
package jp.artan.japanesefoodmod.common.potion

import jp.artan.japanesefoodmod.JapaneseFoodMod
import jp.artan.japanesefoodmod.init.Init.Companion.POTIONS
import net.minecraft.potion.Potion

class JapanesePotion(private val potionName: String, isBadEffectIn: Boolean, liquidColorIn: Int) : Potion(isBadEffectIn, liquidColorIn) {

    init {
        this.setPotionName(this.potionName)
        this.setRegistryName(JapaneseFoodMod.MODID, this.potionName)

        POTIONS.add(this)
    }

    override fun getName(): String {
        return this.potionName
    }
}